### PR TITLE
tpm 3.1.0 (new formula)

### PR DESCRIPTION
**Pull request by iveahugeship**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/tpm.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  def caveats
+    <<~EOS
+      To initialize TPM add this to your tmux configuration file
+      (~/.tmux.conf or $XDG_CONFIG_HOME/tmux/tmux.conf):
+        run '/opt/homebrew/share/tpm/tpm'
+    EOS
+  end
+
+  test do
+    assert_match "", shell_output("#{pkgshare}/tpm")
```
Comment (SMillerDev) [inline Formula/tpm.rb:23]: Can you check for some actual output?

File: Formula/tpm.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  def install
+    pkgshare.install Dir["*"]
+  end
+
+  def caveats
+    <<~EOS
+      To initialize TPM add this to your tmux configuration file
+      (~/.tmux.conf or $XDG_CONFIG_HOME/tmux/tmux.conf):
+        run '/opt/homebrew/share/tpm/tpm'
```
Comment (SMillerDev) [inline Formula/tpm.rb:18]:
```suggestion
        run '#{pkgshare}/tpm'
```

**Review by carlocab — COMMENTED** [review]
Oops, forgot to submit my pending review.

File: Formula/tpm.rb (hunk 2 above)
Comment (carlocab) [reply Formula/tpm.rb:18]: We should use `opt_pkgshare` so we don't rely on a Cellar path

File: Formula/tpm.rb (hunk 1 above)
Comment (iveahugeship) [reply Formula/tpm.rb:23]: Actual output is nothing x) I check if it run without error.

File: Formula/tpm.rb (hunk 2 above)
Comment (iveahugeship) [reply Formula/tpm.rb:18]: So, what should I use? `pkgshare` or `opt_pkgshare`? I don't see this variable [here](https://docs.brew.sh/Formula-Cookbook)

**Review by SMillerDev — APPROVED** [review]

**Comment by iveahugeship** [comment]
@carlocab @carlocab can someone merge this branch please?

**Comment by Moisan** [comment]
Some changes took place in the `Formula` folder since this PR was opened: the formulae were subdivided into subfolders to partition (or shard) them. Could you rebase on master and move this new formula under the `t` folder please?

**Review by chenrui333 — APPROVED** [review]

**Pull request merged**